designs/partition-ui/partition-ui.md

```diff
@@ -0,0 +1,33 @@
+# Partition/region selection UI
+
+## Overview
+Users want the JetBrains toolkit to support other regions outside the classic partition (i.e. China and GovCloud regions).
+
+When a user wants to switch to a different partition, they will need to change their region in addition to their credentials.
+
+This is a quick, point-in-time document with several directions that we can go. Any decisions made here are not one-way doors and we can adjust the experience as necessary
+if there is any user feedback.
+
+## Mockups
```

Comment:
All of these plans unconditionally show all regions from all partitions. Is that manageable from a user's perspective?

Is this PR intended to be a functional spec for the first release, and then make improvements over time, or is the intention to collaborate towards a first design?

Reply:
The desire is to show all partitions/regions only at initial selection phase and if the user wants a region that isn't in their recents list (see screenshot below for this case).

I think this is more of a first release type of thing so customers have something usable and we can make improvements as we get feedback

![image](https://user-images.githubusercontent.com/742829/77594763-9d45fe00-6eb4-11ea-84c4-b2531d5c4328.png)